mapie/calibration.py

```diff
@@ -0,0 +1,559 @@
+        max_class_prob = np.max(pred, axis=1).reshape(-1, 1)
+        y_pred = self.classes_[np.argmax(pred, axis=1)]
+        return max_class_prob, y_pred
+
+    def _check_method(self) -> None:
+        """
+        Check that the method is valid.
+
+        Raises
+        ------
+        ValueError
+            If the method does not belong to the valid methods.
+        """
+        if self.method not in self.valid_methods:
+            raise ValueError(
+                "Invalid method, allowed method are ",
+                (", ").join(self.valid_methods)
+            )
+
+    def _fit_calibrators(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike],
+        calibrator: RegressorMixin,
+    ) -> Dict[int, RegressorMixin]:
+        """
+        This method sequentially fits the calibrators for each labels
+        defined by ``_get_labels`` method.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+        y : ArrayLike of shape (n_samples,)
+            Training labels.
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights for fitting the out-of-fold models.
+            If ``None``, then samples are equally weighted.
+            If some weights are null,
+            their corresponding observations are removed
+            before the fitting process and hence have no residuals.
+            If weights are non-uniform, residuals are still uniformly weighted.
+            Note that the sample weight defined are only for the training, not
+            for the calibration procedure.
+            By default ``None``.
+        calibrator : RegressorMixin
+            Calibrator to fit.
+
+        Returns
+        -------
+        Dict[int, RegressorMixin]
+            Dictionnary of fitted calibrators.
+        """
+        X, y = indexable(X, y)
+        y = _check_y(y)
+        max_prob, y_pred = self._get_labels(X)
+        calibrators = {}
+        for item in np.unique(y_pred):
+            calibrator_ = self._fit_calibrator(
+                item,
+                calibrator,
+                y,
+                max_prob,
+                y_pred,
+                sample_weight,
+            )
+            calibrators[item] = calibrator_
+        return calibrators
+
+    def _pred_proba_calib(
+        self,
+        idx: int,
+        item: int,
+        calibrated_values: NDArray,
+        max_prob: NDArray,
+        y_pred: NDArray,
+        calibrators: Dict[int, RegressorMixin],
+    ) -> NDArray:
+        """
+        Using the predicted scores, we calibrate the maximum score with the
+        specifically fitted calibrator. Note that if there is no calibrator
+        for a the specific class, then we simply output the not calibrated
+        values.
+
+        Note that if the calibrated score prediction is 0, there would be an
```

Comment:
This could happen if in the training sample, there are no correct observations and therefore the calibration procedure would set all the corrected predictions to 0.

Reply:
Please update the docstring, we changed the method